plugin/pkg/shoot/validator/admission.go

```diff
@@ -262,6 +295,41 @@ func (v *ValidateShoot) Admit(a admission.Attributes, o admission.ObjectInterfac
 		allErrs field.ErrorList
 	)
 
+	// Prevent removal of `gardener` finalizer if the shoot still has backing assets, e.g. networks, machines, etc.
+	// This is done by checking if the shoot namespace in the seed exists.
+	if seed != nil && len(shoot.Status.TechnicalID) > 0 {
+		oldFinalizers := sets.NewString(oldShoot.Finalizers...)
+		newFinalizers := sets.NewString(shoot.Finalizers...)
+
+		if oldFinalizers.Has(corev1alpha1.GardenerName) && !newFinalizers.Has(corev1alpha1.GardenerName) {
+			seedSecret, err := v.secretLister.Secrets(seed.Spec.SecretRef.Namespace).Get(seed.Spec.SecretRef.Name)
+			if err != nil {
+				return apierrors.NewInternalError(fmt.Errorf("could not get referenced seed secret: %+v", err))
+			}
+
+			seedClient, err := v.runtimeClientFactory.CreateRuntimeClientFromSecret(seedSecret,
+				kubeclient.WithClientOptions(client.Options{
+					Scheme: kubeclient.SeedScheme,
+				}),
+			)
+
+			if err != nil {
+				return apierrors.NewInternalError(fmt.Errorf("could not create seed client to check backing resources: %+v", err))
+			}
+
+			ns := &v1.Namespace{}
+			err = seedClient.Get(context.Background(), kutil.Key(shoot.Status.TechnicalID), ns)
```

Comment:
```suggestion
			err = seedClient.Get(context.TODO(), kutil.Key(shoot.Status.TechnicalID), ns)
```

Probably it's safer to use `context.TODO()` until we have proper context handling for the admission plugins. The background context is never canceled.